File: leetcode/283.move-zeroes/main.go
package main

import "fmt"

func moveZeroes(nums []int) {
	l := len(nums)
	for i := 0; i < l; i++ {
		if nums[i] == 0 {
			j := i + 1
			for j < l && nums[j] == 0 {
				j++
			}
			if j < l {
				nums[i], nums[j] = nums[j], nums[i]
			} else {
				break
			}
		}
	}
}

func main() {
	a := []int{0, 1, 1, 0, 1, 1}
	moveZeroes(a)
	fmt.Println(a)
}

File: leetcode/283.move-zeroes/main.rs
struct Solution;

impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let (mut left, mut right, length) = (0, 0, nums.len());
        while right < length {
            if nums[right] != 0 {
                nums.swap(left, right);
                left += 1;
            }
            right += 1;
        }
    }
}

fn main() {
    let mut nums = vec![0, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0];
    Solution::move_zeroes(&mut nums);
    println!("{:?}", nums);
}
